### PR TITLE
DB-10570 support currentSchema in jdbc connecting string

### PR DESCRIPTION
**Pull request by yxia92**

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java [hunk 1]
```diff
@@ -173,7 +172,9 @@
         drdaID = info.getProperty(Attribute.DRDAID_ATTR, null);
         rdbIntTkn = info.getProperty(Attribute.RDBINTTKN_ATTR, null);
         ipAddress = info.getProperty(Property.IP_ADDRESS, null);
-        defaultSchema = info.getProperty(Property.CONNECTION_SCHEMA, null);
+        defaultSchema = info.getProperty(Property.CONNECTION_CURRENT_SCHEMA, null);
+        if (defaultSchema == null)
+            defaultSchema = info.getProperty(Property.CONNECTION_SCHEMA, null);
```
Comment (ascend1) [inline db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:177]: It seems we support both `schema` and `currentSchema` then and `currentSchema` has priority in a sense that if it's set, then `schema` is ignored. Is that intended?

**Review by dgomezferro — APPROVED** [review]

**Review by arnaud-splice — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java (hunk 1 above)
Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:177]: e.g. what's the expected behavior of `schema=A;currentSchema=B`.

**Review by hatyo — CHANGES_REQUESTED** [review]
LGTM, but I think we should address @ascend1's comment.

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java (hunk 1 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:177]: Yes, @ascend1 and @hatyo , the code above is intended, so for the scenario @hatyo raised, we have a deterministic behavior. Basically, currentSchema will take precedence over schema.

**Review by carolp-503 — APPROVED** [review]
+1

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java (hunk 1 above)
Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:177]: Thanks!

**Review by hatyo — APPROVED** [review]

**Review by ascend1 — APPROVED** [review]

**Pull request merged**